docs/libcurl/opts/CURLOPT_SSL_OPTIONS.md

```diff
@@ -86,6 +86,16 @@ certificate that supports client authentication in the OS certificate store it
 could be a privacy violation and unexpected.
 (Added in 7.77.0)
 
+## CURLSSLOPT_EARLYDATA
+
+Tell libcurl to try sending application data as TLS1.3 early data. This option
+is only supported for OpenSSL, WolfSSL, MbedTLS and GnuTLS. This option works
+on a best effort basis, in cases when it wasn't possible to send early
+data the request is resent normally post-handshake.
+This option does not work when using QUIC or HTTP/2, it only works when using
+HTTP/1.1 without a proxy, otherwise its a no-op.
+(Added in 8.10.0)
+
```

Comment:
How is this limited to HTTP/1 only? I don't see any changes in the HTTP code, so why does it limit itself to just version 1 and not to other versions or other protocols?

Reply:
It kind of works with http/2 but it doesn't maximally utilise sending early data, it sends the initial http/2 frames as early data and the rest of them are sent post handhshake. https://github.com/curl/curl/pull/14135#issuecomment-2230520071, maximally utilising sending early data would require significant refactoring of `multi_runsingle` https://github.com/curl/curl/pull/14135#issuecomment-2230536121

http/3 doesn't use the `Curl_cft_ssl` cfilter hence it isn't supported by this PR

The actual behaviour is that only the first tls record sent over the connection (as part of any protocol) is sent as early data, the rest is sent post handshake.